src/GitVersionTask/FileHelper.cs

```diff
@@ -63,17 +63,13 @@ public static void DeleteTempFiles()
 
         public static string GetFileExtension(string language)
         {
-            switch (language)
+            return language switch
             {
-                case "C#":
-                    return "cs";
-                case "F#":
-                    return "fs";
-                case "VB":
-                    return "vb";
-                default:
-                    throw new ArgumentException($"Unknown language detected: '{language}'");
-            }
```

Comment:
Unrelated to your changes, but I really don't fancy the `FileHelper` class. Seems like at least one or perhaps several missing abstractions it's attempting to fill. The `<Language>FileContainsVersionAttribute` methods are screaming for an interface with different implementations for each language, for instance. Not sure we need to tackle that in this PR, but we should some day.

Reply:
Agree, in another PR